odoo/addons/base/views/res_country_views.xml

```diff
@@ -23,7 +24,7 @@
                     <sheet>
                         <div class="oe_button_box">
                         </div>
-                        <field name="image" widget="image" class="oe_avatar"/>
+                        <field name="image_url" widget="image_url" class="oe_avatar"/>
```

Comment:
1. 
It's showing an error when opening the form when the flag is not available:

![image](https://user-images.githubusercontent.com/41282536/91956868-8fb75e80-ed05-11ea-9947-fe7fa35101a5.png)

Is that intended?

2. 
Before, when adding a country, I was able to upload an image for it.
Obviously the new implementation does not allow any upload, but shouldn't I be able to specify an image path if I want to add my own countries?
Same thing for the edition, would be nice to allow overriding the path to customize images as necessary.

Reply:
1) It's caused by the wrong_url above and will be removed.
2) The countries are not something which should be created manually...
IMO, this should be handled by code update & upgrade (if one day a country is created), and the flags are static are not modified frequently either...

Reply:
I agree that adding a country is a fix that will be done in stable, adding the necessary file ... custom (imaginary ?) countries is not a core feature :D .

Reply:
I agree with you completely.
But then again we have a big green link that says "Add other countries"

![image](https://user-images.githubusercontent.com/41282536/91960854-b1ffab00-ed0a-11ea-8cdc-f642de93da0a.png)

So I was just asking to make sure ^^

Reply:
Mmmh, strange to have a link to add countries. The kind of data that people should not really mess with ... anyway indeed we could support both a url and image fields. If we give an image, url could be computed like web/image/country/image . That way we support both use cases, and standard countries have no footprint in db. What do you think @Feyensv  ? Additional point: no need of migration :D .

But I fill like this is overkill.

Reply:
Could be overkill, could also avoid people yelling at us when migrating because we broke all their super nice imported images of countries in 4k resolution.

Not sure, would be cool to have some statistics about who changes that on the saas but I don't how if it's possible / easy.

Reply:
Or we store the url field, allowing to change it and link to a hosted image if you want to. That way we still don't hold unnecessary data in db, and let people host their own 4K images for Wakanda.

Reply:
It was Odo request to not store the url field (it was stored first)...
IMO, adding countries and/or modifying the flag doesn't have to be supported...

Reply:
Then this link in my image above is super confusing.
Maybe check with the Sales/Stock PO (good luck finding who that is) why it is there and remove it in your PR?

Reply:
It is FP :D . Maybe just ask LNA for confirmation, not sure we need to escalate ;) .

For stored field: if Odo does not want it, well ... do you have a link to the discussion ?

Reply:
For information, in res.lang we now use the country flag by default, if you don't have overrided the res.lang flag_image field, flag_image_url is a computed field not stored...

Purpose was to use static path instead of /web/image/res.country/flag_image... to be served by nginx and not werkzeug.

Reply:
Thanks for the info !

Could be great to find the country and use its image_url value if no custom image is given, so that we don't manually handcraft too much country_flags URL .

Reply:
> For stored field: if Odo does not want it, well ... do you have a link to the discussion ?

After research, I cannot find a message in itself, thus I suppose it was said in voice chat :roll_eyes:.
Should I make a two-fields logic like `res.lang` ?

> It is FP :D . Maybe just ask LNA for confirmation, not sure we need to escalate ;) .

I'll update the link title (and the ACL ?)